src/Microsoft.PowerShell.Commands.Utility/commands/utility/AddType.cs

```diff
@@ -551,7 +551,21 @@ protected override void BeginProcessing()
             {
                 ThrowTerminatingError(
                     new ErrorRecord(
-                        new PSNotSupportedException(AddTypeStrings.CannotDefineNewType), "CannotDefineNewType", ErrorCategory.PermissionDenied, null));
+                        new PSNotSupportedException(AddTypeStrings.CannotDefineNewType),
+                        nameof(AddTypeStrings.CannotDefineNewType),
+                        ErrorCategory.PermissionDenied,
+                        targetObject: null));
+            }
+
+            // 'ConsoleApplication' and 'WindowsApplication' types are currently not working in .NET Core
+            if (OutputType != OutputAssemblyType.Library)
```

Comment:
Can we ask the API owner? Is this a bug in the cmdlet, in the API or it is "by design"?

Reply:
See my comment at https://github.com/PowerShell/PowerShell/issues/13344#issuecomment-674593738

Reply:
Those 2 output kinds don't work, and they have been this way since 6.0, so making them throw terminating error makes sense. This doesn't block someone from fixing them later.